assets/aml-benchmark/components/src/batch_output_formatter/result_converters.py

```diff
@@ -4,85 +4,158 @@
+                del usage[old_key]
+        usage['batch_size'] = result.get('batch_size', 1)
+        if self._model.is_oss_model():
+            usage['input_token_count'] = self._get_oss_input_token(usage)
+            usage['output_token_count'] = self._get_oss_output_token(result, usage)
         return usage
 
     def convert_result_ground_truth(self, result: Dict[str, Any]) -> Dict[str, Any]:
         """Convert the result to ground truth."""
         ground_truth = ''
-        if self._is_llama_model():
+        use_ground_truth_input = False
+        if self._model.is_oss_model():
             if self._metadata_key:
                 ground_truth = self._get_request(result)[self._metadata_key][self._label_key]
-            else:
+            elif self.METADATA_KEY_IN_RESULT in result:
                 ground_truth = result[self.METADATA_KEY_IN_RESULT][self._label_key]
-        elif self._is_aoai_model():
+            else:
+                use_ground_truth_input = True
+        elif self._model.is_aoai_model():
+            use_ground_truth_input = True
+        if use_ground_truth_input:
             for k, v in self._lookup_dict.items():
-                if k in self._get_request(result):
+                if k in self._get_request_content(result):
                     ground_truth = v
         return {'ground_truth': ground_truth}
 
     def _get_raw_output(self, result: Dict[str, Any]) -> Dict[str, Any]:
         prediction = ''
-        if self._is_llama_model():
+        if self._model.is_oss_model():
             prediction = ResultConverters._get_oss_response_result(result)
-        elif self._is_aoai_model():
+        elif self._model.is_aoai_model():
             prediction = ResultConverters._get_aoai_response_result(result)
-        return {'prediction': prediction}
+        return {ResultConverters.PREDICTION_COL_NAME: prediction}
+
+    def _get_fallback_output(self, is_perf: bool = False) -> Dict[str, Any]:
+        if is_perf:
+            result = {k: -1 for k in ResultConverters.PERF_OUTPUT_KEYS}
+            result['start_time_iso'] = ResultConverters.DEFAULT_ISO_FORMAT
+            result['end_time_iso'] = datetime.datetime.utcnow().isoformat()
+        return {ResultConverters.PREDICTION_COL_NAME: self._fallback_value}
+
+    def is_result_success(self, result: Dict[str, Any]) -> bool:
+        """Check if the result contains a successful response."""
+        if result['status'].lower() != "success":
+            return False
+        # handle the scenario the 200 with failure in response.
+        try:
+            _ = self._get_raw_output(result)
+        except Exception as e:
+            logger.warning(f'Converting meet errors {e}')
+            return False
+        return True
+
+    def _get_oss_input_token(self, perf_metrics: Any) -> Tuple[int, int]:
+        if self._is_performance_test:
+            return ResultConverters.DEFAULT_PERF_INPUT_TOKEN
+        return perf_metrics.get('input_token_count', -1)
```

Comment:
In various cases, -1 has been used as the default value if token count is missing.
1. Is there a scenario where the key will be missing?
2. If we don't have the token count, can we just skip it from the output. Otherwise in compute_performance_metrics component, it will give negative values in calculations.

Reply:
1. For llama without perf. Or AOAI failed but customer choose to use fallback.
2. I will remove those.